src/libsyntax/parse/token.rs

```diff
@@ -365,6 +365,10 @@ impl Token {
         self == &Lt || self == &BinOp(Shl)
     }
 
+    crate fn is_args_start(&self) -> bool {
+        self.is_qpath_start() || self == &OpenDelim(Paren)
```

Comment:
Well, the tokens technically coincide with the `is_qpath_start` set, but `<`/`<<` in `Type<T>`/`Type<<X as Y>::Assoc>` is not a qpath start at all.
I'd remove this commit.